project-codeflare/codeflare-sdk
# Remove creation of OAuth resources/logic and remove openshift_oauth option

# Issue link


Issue: https://issues.redhat.com/browse/RHOAIENG-1993 

# What changes have been made

- This PR removes the logic in the SDK that creates the OAuth resources.
- Remove `openshift_oauth` from ClusterConfiguration as defaulting to True.
- Added `COOKIE_SECRET` env for oauth proxy container. - [Here](https://github.com/project-codeflare/codeflare-operator/pull/453#:~:text=%2D%20%2D%2Dcookie%2Dsecret%3D%24(COOKIE_SECRET))
- Removed unit tests related to the creation/deletion of OAuth resources.
- Adjusted test cases and unit tests.
- Verified this change works with the RayCluster Controller
- Add `verify_tls` option to ClusterConfiguration: this option is set to True by default but still has the option to change for debugging. Based on this option the endpoint will be verified, if set to false, it alters if the user verifies that the certificate is trusted.

# Verification steps



To test these changes:
0. In an OpenShift cluster.
1. Checkout this PR [RayCluster Controller](https://github.com/project-codeflare/codeflare-operator/pull/453). In the CFO repository you can run the CFO in your cluster with `make run NAMESPACE=default` and `make install`.
2. Deploy and install KubeRay
3. Checkout this PR and install the SDK
4. Run through [this notebook](https://github.com/project-codeflare/codeflare-sdk/blob/main/demo-notebooks/guided-demos/2_basic_jobs.ipynb), change between verify_tls=True/False.
5. Once `cluster.up()` is ran, the oauth-proxy sidecar is created by default if on an OpenShift cluster. You should find several resources created including a ServiceAccount named `jobtest-oauth-proxy` and a route named `ray-dashboard-jobtest` which were created by the RayCluster Controller as opposed by the SDK.
6. The route should be accessible through authentication.

## Checks
- Testing Strategy
